src/user/use-cases/sign-in/sign-in.ts

```diff
@@ -1,42 +1,35 @@
 import { Injectable, UnauthorizedException } from '@nestjs/common';
-import { JwtService } from '@nestjs/jwt';
 
+import { IUserRepository } from '@/user/interfaces';
+import { JwtAuthProvider } from '@/auth/providers';
 import { SaveUserDTO } from '@/user/dto';
-import { IDecodedTokenPayload, IUserRepository } from '@/user/interfaces';
 
 @Injectable()
 export class SignInUseCase {
   constructor(
     private usersRepository: IUserRepository,
-    private jwtService: JwtService,
+    private jwtAuthProvider: JwtAuthProvider,
   ) {}
 
-  private async decodeToken(jwtToken: string): Promise<SaveUserDTO> {
-    const secret = process.env.AUTH0_CLIENT_SECRET;
+  async execute(token: string): Promise<{ status: number; message: string }> {
+    let userPayload: SaveUserDTO;
 
     try {
-      const { sub, name, email } =
-        await this.jwtService.verifyAsync<IDecodedTokenPayload>(jwtToken, {
-          secret,
-        });
-
-      const userPayload: SaveUserDTO = {
-        id: sub,
-        name,
-        email,
-      };
-
-      return userPayload;
-    } catch (err) {
-      throw new UnauthorizedException('Token inválido.');
+      userPayload = await this.jwtAuthProvider.decodeToken(token);
+    } catch (err: any) {
+      throw new UnauthorizedException({
+        message: 'Token inválido.',
+        reason: err.message,
+      });
     }
-  }
-
-  async execute(token: string): Promise<void> {
-    const userPayload = await this.decodeToken(token);
 
     const user = await this.usersRepository.findByEmail(userPayload.email);
 
-    if (!user) await this.usersRepository.create(userPayload);
+    if (!user) {
+      await this.usersRepository.create(userPayload);
```

Comment:
Por que essas chamadas do repositorio nao estao no try catch? Como o resto da logica necessita do que esta dentro do try catch nao faz sentido ficar fora, alem de prevenir um possivel erro no repository, assim vc nao precisa declarar o `let userPayload: SaveUserDTO;` antes do try, pode deixar `const userPayload = await this.jwtAuthProvider.decodeToken(token);`